pkg/daemon/server/service/runtime/pod_tracker.go

```diff
@@ -0,0 +1,125 @@
+package runtime
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+	"github.com/numaproj/numaflow/pkg/shared/util"
+	"go.uber.org/zap"
+)
+
+// podInfoSeparator is used as a separator to split the pod key
+// to get the pipeline, vertex and pod index.
+// "*" is chosen because it is not allowed in the above fields.
+const podInfoSeparator = "*"
+
+type PodTracker struct {
+	pipeline        *v1alpha1.Pipeline
+	log             *zap.SugaredLogger
+	httpClient      monitorHttpClient
+	activePods      *util.UniqueStringList
```

Comment:
It can be `map[string]int` -> vertexname to the max replica or` map[string][]int`